core/arch/arm/kernel/tee_ta_manager.c

```diff
@@ -115,6 +116,55 @@ static void set_tee_rs(struct tee_ta_session *tee_rs)
 	thread_set_tsd(tee_rs);
 }
 
+static bool tee_ta_try_set_busy(struct tee_ta_ctx *ctx,
+			struct tee_ta_session *sess)
+{
+	bool rv = false;
+
+	mutex_lock(&tee_ta_mutex);
+
+	/*
+	 * If we're about to set a context of a single instance TA busy
+	 * make sure that we don't have any other single instance TA in the
+	 * call stack to guarantee that we can't deadlock.
+	 */
```

Comment:
I was thinking about actual deadlock detection, something like: if `ctx->busy` is true, and `ctx` is single instance TA, and `ctx` is already waiting for some single instance TA that we have in the current stack, then we are about to deadlock, otherwise it's OK.
Now, assuming the above condition is correct, I'm not sure it is worth the trouble (and the associated overhead).